[Feature] LoginView, HomeView 정렬 기능, Font System 추가

kipsong133 opened the pull request:
## 📌 배경

close #39 
close #41 
close #18 

## 내용

- LoginView UI 구성
- HomeView 날짜순 정렬 기능
- FontSystem + 주석(size / lineHeight / FontWeight 확인용이하도록 주석 추가)
(참고로 아이콘은 TestFlight 제출을 위해, Dummy 이미지로 추가했습니다.)
- 햅틱 진동
- 태그리스트 UI 수정

## 테스트 방법 (optional)

- TestFlight 앱 실행 (1.0.5)

## 스크린샷 (optional)

<img width="475" alt="image" src="https://user-images.githubusercontent.com/65879950/207056792-1dfa962d-0607-4c73-a05f-abf9a04c22b2.png">

<img width="475" alt="image" src="https://user-images.githubusercontent.com/65879950/207056828-b41f7ccf-4926-4bfe-adfe-ab77783a59ac.png">

kipsong133 committed:
feat: 폰트 시스템 구성

- #39
- DesignSystem 중 Font System(Pretendard) 구성

kipsong133 committed:
feat: dummy data로 월별 정렬 기능 구현

- #41

kipsong133 committed:
feat: LoginView UI 구성

kipsong133 committed:
Merge branch 'develop' of https://github.com/depromeet/nyongnyong_iOS into develop

Marcorable commented on Targets/UserInterface/Sources/Home/HomeView.swift:
```diff
@@ -8,31 +8,32 @@
-    init() {
-//        UINavigationBar.appearance().largeTitleTextAttributes = [.font : UIFont(name: "Pretendard-Bold", size: 14)!]
-    }
+    @StateObject var viewModel = HomeViewModel()
```
`@StateObject` 를 사용하신 이유가 있으신지 궁금합니다!

kipsong133 replied:
StateObject 는 화면을 다시 그릴 때, 변경된 부분 외에 다시 그리지 않아서 사용하긴 하는데,
혹시 ObservedObject 랑 StateObject 중 선택에 대한 이유가 궁금하신 건가요? 혹은 다른이유일까요!?

Marcorable replied:
말씀해주신 두 가지 중 선택한 이유가 궁금했습니다!

탭바 변동만 생각하고 View가 특별히 init 되지 않을꺼 같다고 생각하고 질문드렸는데..
지금 다시 생각해보니.. HomeView나 CommunityView 하위뷰에서 확실히 View 변동이 생기고 init 될거 같아요..

답변 감사합니다!

Marcorable commented on Targets/UserInterface/Sources/Home/HomeView.swift:
```diff
@@ -8,31 +8,32 @@
             mainBody()
                 .toolbar(content: {
                     // 네비게이션 제목
                     ToolbarItemGroup(placement: .navigationBarLeading) {
+                        // TODO: 이미지로 변경 필요
                         Text("TICLEMOA")
-                            .customFont(14, .bold)
+                            .pretendFont(.title3)
                     }
                     
                     // 종모양 & 마이프로필
                     ToolbarItemGroup(placement: .navigationBarTrailing) {
                         NavigationLink(
                             destination: AlarmView(),
                             label: {
-                                Image("alarm")
+                                // TODO: 이미지 에셋 추가 시, 등록
+//                                Image("alarm")
+                                Image(systemName: "bell.badge")
+                                    .symbolRenderingMode(.palette)
+                                    .foregroundColor(.primary)
                             })
                         NavigationLink(destination: ProfileView(), label: {
                             Image("defaultProfile")
-                            
                         })
                     }
                 })
```
mainBody 네이밍이 불 필요하다고 느껴집니다..

`ToobarItemGroup` 별로 연산프로퍼티로 분리하고 (`@ViewBuilder`) 
-> 주석 대신 프로퍼티 네이밍으로 용도 전달하는 형태로 구성하면 어떨까 제안해봅니다!

kipsong133 replied:
넵. 함수보다는 연산프로퍼티로 사용해보죠~!

Marcorable replied:
[아래 답변](https://github.com/depromeet/nyongnyong_iOS/pull/46#discussion_r1046546633)과 엮어서 
같이 고민해보고 스타일을 정하면 좋을 것 같습니다!

Marcorable commented on Targets/UserInterface/Sources/Login/LoginView.swift:
```diff
@@ -0,0 +1,100 @@
+            .setupBackground()
+    }
+    
+    func mainBody() -> some View {
+        VStack {
+            Spacer()
+                .frame(maxHeight: 61)
+            headerTitles
+            Spacer()
+            Color.grey2
+                .frame(maxWidth: 214, maxHeight: 214)
+            
+            Spacer()
+            socialLoginButtons
+            Spacer()
+                .frame(maxHeight: 58)
+        }
+    }
```
이 부분도 저는 위에 언급한 `mainBody()`와 같이 불필요한 구성이란 생각을 가지고 있는데..

혹시 이렇게 구성하신 방향이 어떤 의도인지 궁금합니다!

kipsong133 replied:
UI 를 나눌 때, 3 가지 정도로 그냥 나누고 싶었습니다.
1. 화면 전체에 적용되는 UI -> mainBody() 호출부에 추가
2. 화면의 위치에 따른 배치 -> mainBody 바로 하위에 추가
3. 각각의 세부구현 -> View 로 생성 혹은, 연산 프로퍼티

사실 이건 저만의 기준으로 작성한거라 의견 주시면, 감사하겠습니다 ~! : )

Marcorable replied:
[위 질문](https://github.com/depromeet/nyongnyong_iOS/pull/46#discussion_r1045868528) 과 엮인것 같네요..
> 약간 야곰 가르침 탓에 습관적인 부분의 스타일인데....
> 함수/변수 네이밍 + 인자 레이블 등을 이용해서 최대한 코드를 가독성 좋게 만들고
> 주석은 최소화 (기능 세부사항) 스타일을 지향하는 편이었습니다

확실히 스타일이라 느껴지는 부분인거 같아서..
함께 맞춰가면 좋을것 같다고 생각이 드네요!

약간 저는 우성님이 말씀해주신 상황에서 생각에

1. 화면 전체에 적용되는 UI -> 부모 뷰에서 호출/할당 부에서 ViewModifier 지정 
       + NavigationView 등 특수한 경우 전체 ViewModifier body 에 적용
2. 화면의 위치에 따른 배치 -> body에 배치
3. 각각의 세부구현 -> View 로 생성 혹은, 연산 프로퍼티

형태로 작성 했었거든요.

맨날.. 소환하는 것 같은데.. 죄송합니다 ㅎㅎㅎ
@joseph704 @sustainable-git 

함께 고민해보고 Ticlemoa 프로젝트에서 스타일을 맞추면 어떨까 생각합니다!

kipsong133 replied:
혹시 이부분 예제코드가 있을까요? 제가 어떤식으로 할지 잘 그림이 안그려집니다.ㅠㅠ

sustainable-git replied:
@keeplo 

저는 주로 fileprivate struct를 만드는 편이어서, 함수 형태로 UI를 작성하는게 새롭긴 합니다.
View가 함수 형태로 만들어진다면 UI 구성과 Action의 구분요소가 하나 줄어드니 가독성이 떨어지지 않나 생각이 들긴 하네요.

sustainable-git replied:
@keeplo 

저는 재사용할 일이 있지 않으면 `var body` 구문에 다 만드는 편이라
확실히 mainBody() 구문이 불필요하게 느껴지긴 하네요.

Marcorable reviewed (changes requested):
잔잔바리로 질문과 제안을 코멘트로 몇가지 남겼습니다..

왜 안되지? 10번 외치고 정복.. 멋집니다.. 고생하셨습니다...
폰트 시스템... 너무 고생많으셨어요..

kipsong133 committed:
fix: kakao library 컴파일 에러 수정 및 PR 코멘트 적용

- kakao Library 코드에 에러가 있어서 수정
- PR 코멘트 피드백 (기존 Method 로 구성한 "mainBody()" 를 연산프로퍼티 "mainbody" 로 수정)

sustainable-git commented on Targets/UserInterface/Sources/ContentView.swift:
```diff
@@ -7,14 +7,29 @@
+    @State private  var isLoggedIn: Bool = false
+    
+    public init() {
+        let appKey = "a864acaf7534a2c1a627e21d352d2e49"
```
api key를 가릴 방법이 필요할것 같습니다.

kipsong133 replied:
앱 배포 단계에서는 "Config.xcconfig" 을 활용해서 Info.plist 에서 접근하도록 하게하려고해요.

물론 gitIgnore 에 해당 파일은 추가해서, 따로 전달드리는 식으로 해야할 것 같구요!

kipsong133 committed:
fix: Font system 주석 추가

- 사용 시, 주석을 통해 lineHeight, size, fontWeight 확인할 수 있도록 주석 추가

kipsong133 committed:
feat: HomeView 태그리스트 제스쳐 및 헤더 수정

- HomeView 테그리스트 UI 제스처 추가 및 UI 퍼블리싱 미흡한 부분 수정 (Padding, Color ..etc)
- Sticky Header 동작부분 제거 및 DataFormatter 수정

kipsong133 committed:
feat: 햅틱 진동 추가

- HapticsManager 를 통한 햅틱진동관리 객체 추가
- 로그인 버튼 및 태그 선택 버튼에 추가
#63

kipsong133 committed:
Merge branch 'develop' of https://github.com/depromeet/nyongnyong_iOS into develop

# Conflicts:
#	Targets/UserInterface/Sources/Home/HomeView.swift

kipsong133 committed:
fix: iOS 15 대응

- TextField iOS 16 버전 컴포넌트 이전 버전 TextField 로 적용
- PresentationDetent iOS 16 한정 사용하도록 프로퍼티 래퍼 추가

kipsong133 committed:
fix: NavigationTitle 위치 수정

- HomeView 에 있는 ToolBarItem 을 MainTabView 로 이동

joseph704 commented:
확인했습니다. 고생하셨습니다!

kipsong133 committed:
feat: Kakao Login API 연동

- API Test 만 마친 상태로, UI 업데이트는 미반영상태

kipsong133 committed:
feat: KakaoLogin API UI 연동

- Kakao Login SDK 연결
- 이후 TickleMoa 서버에 OAuth Token 전달

kipsong133 committed:
feat: Apple Login UI 구현

kipsong133 committed:
feat: AccessToken 싱글톤으로 저장

kipsong133 committed:
fix: guard 문 조건 수정

- error 가 nil 일 때 통과하도록 수정
(기존에 nil이 아니면 통과하도록하여서, 에러일 떄만 통과하도록 코드 작성했음)

kipsong133 committed:
Merge branch 'develop' of https://github.com/depromeet/nyongnyong_iOS into develop

# Conflicts:
#	Targets/UserInterface/Sources/MyPage/MyPageView.swift

kipsong133 committed:
feat: 아티클 추가 API 구현

kipsong133 committed:
Merge branch 'login-network' into develop

# Conflicts:
#	Targets/UserInterface/Sources/MyPage/MyPageView.swift
#	Targets/UserInterface/Sources/Utils/Extensions/PresentationDetent+.swift

Marcorable commented on Targets/UserInterface/Sources/Home/HomeArticle/Article.swift:
```diff
@@ -0,0 +1,153 @@
+
+import Foundation
+import Collections
+import UIKit
```
혹시 UIKit import가 안되면 에러가 발생할까요?

kipsong133 replied:
다른 작업하다가 지우지 못했나봅니다~! 감사합니다!

Marcorable reviewed (approved):
고생하셨습니다

kipsong133 committed:
Merge branch 'develop' into develop

kipsong133 committed:
Merge branch 'develop' of https://github.com/kipsong133/nyongnyong_iOS into develop

# Conflicts:
#	Targets/UserInterface/Sources/MainTab/MainTabView.swift
#	Targets/UserInterface/Sources/MyPage/MyPageView.swift

kipsong133 committed:
mod: customFont -> pretendFont 로 수정

- 폰트시스템 적용한 Modifier로 수정

kipsong133 commented:
Rebase 로 몇 번이고 해봤는데, 각각의 PR들이 컴파일에러가 많아서, 소통이 더어려워 질 수도 있을 것 같습니다.
제가 급하다는 핑계로 너무 커뮤니케이션 부분 신경안쓰고 코드 쓴 것 같습니다 ㅠㅠ.

이번만 Squash 하겠습니다~!
